### PR TITLE
Solver solver

### PR DESCRIPTION
**Pull request by jiayuanz3**

### CODE REVIEW

File: epistrains/solver.py [hunk 1]
```diff
@@ -1,3 +1,34 @@
...
+        defaults to 1
+    :type time: float, optional
+    """
+    def __init__(self, strains: List[Strain], t_eval, time = 1):
+        self.time = round(time, 2)
+        self.solution = []
+        self.strains = strains
+        self.t_eval = t_eval
+
+    def solver(self):
```
Comment (Ellmen) [inline epistrains/solver.py:21]: Methods should be verbs (so solve not solver)

File: .github/workflows/python-app.yml [hunk 2]
```diff
@@ -27,6 +27,7 @@ jobs:
       run: |
         python -m pip install --upgrade pip
         pip install flake8 pytest pytest-cov
+        pip install .
```
Comment (Ellmen) [inline .github/workflows/python-app.yml:30]: Do we actually need this?

Comment (rccreswell) [reply .github/workflows/python-app.yml:30]: This is required as it pip installs the dependencies we have for the package in the setup.py

Comment (Ellmen) [reply .github/workflows/python-app.yml:30]: Oh the dependencies are aren't in a requirements.txt and we're actually using them now

File: epistrains/solver.py [hunk 3]
```diff
@@ -1,3 +1,33 @@
...
+    :type strains: _type_
+    :param t_eval: a list of arguments for start time, end time and time interval
+    :type t_eeval: list
+    :param time: timeframe (h) over which the models should be solved for, 
+        defaults to 1
+    :type time: float, optional
+    """
+    def __init__(self, strains: List[Strain], time = 1):
+        self.time = round(time, 2)
+        self.solution = []
```
Comment (Ellmen) [inline epistrains/solver.py:18]: I think this should default to `None` to make it clear that it hasn't been solved yet. I'll probably make a new PR soon with some style changes though, so this is fine for now.

**Review by Ellmen — APPROVED** [review]

**Pull request merged**